app/src/nxt/java/com/pileproject/drive/programming/visual/block/selection/IfNxtWasTouchedBlock.java

```diff
@@ -31,15 +33,22 @@
  * @version 1.0 7-July-2013
  */
 public class IfNxtWasTouchedBlock extends SelectionBlock {
-
+    private String mFirmware;
     public IfNxtWasTouchedBlock(Context context) {
         super(context);
         LayoutInflater.from(context).inflate(R.layout.block_if_nxt_was_touched, this);
+
+        mFirmware = MachinePreferences.get(getContext()).getFirmware();
     }
 
     @Override
     public int action(MachineController controller, ExecutionCondition condition) {
-        condition.pushSelectionResult(((NxtController) controller).getTouchSensorValue());
+        boolean wasTouched = ((NxtController) controller).getTouchSensorValue();
+
+        // leJOS returns the opposite value
+        if (mFirmware.equals(MachinePreferencesSchema.FIRMWARE.LEJOS)) wasTouched = !wasTouched;
```

Comment:
Since mFirmware never changes, can it replace with <code>boolean</code> in the constructor?
(though recent JVM might find this and optimize it)